src/figuresClasses.ts

```diff
@@ -1,11 +1,71 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side lengths must be greater than 0');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error(`Sides ${a}, ${b}, and ${c} can't form a triangle`);
```

Comment:
[CODE STYLE] - The error message is good but could be enhanced by specifying the triangle inequality rule explicitly. For example: 'Triangle side lengths do not satisfy the triangle inequality theorem: a=${a}, b=${b}, c=${c}'